Core/StitchCoreSDK/Sources/StitchCoreSDK/Auth/Internal/CoreStitchAuth+AuthStorage.swift

```diff
@@ -0,0 +1,146 @@
+        let authInfoAny = storage.value(forKey: activeUserStorageKey)
+
+        guard let authData = authInfoAny as? Data else {
+            return nil
+        }
+        return try JSONDecoder().decode(StoreAuthInfo.self, from: authData).toAuthInfo
+    }
+
+    /**
+     * Writes an `AuthInfo` struct into some underlying storage.
+     *
+     * - parameters:
+     *     - activeAuthInfo: The `AuthInfo` to write to `Storage`
+     *     - toStorage: The `Storage` to which to write the `AuthInfo`.
+     * - throws: if the `AuthInfo` could not be encoded into JSON.
+     */
+    internal func writeActiveUserAuthInfoToStorage(activeAuthInfo authInfo: AuthInfo,
+                                                   toStorage storage: Storage) throws {
+        storage.set(try JSONEncoder().encode(StoreAuthInfo.init(withAuthInfo: authInfo)),
+                    forKey: activeUserStorageKey)
+    }
+
+    /**
+     * Reads an Array of `AuthInfo` from some underlying storage.
+     *
+     * - parameters:
+     *     - fromStorage: The `Storage` from which to read the `AuthInfo` array.
+     * - throws: if the auth information in the underlying storage is corrupted or missing.
+     * - returns: A List of [`AuthInfo`] classes.
+     */
+    internal func readCurrentUsersAuthInfoFromStorage(fromStorage storage: Storage) throws -> [AuthInfo] {
+        var authInfos: [AuthInfo] = []
+        let authInfoAny = storage.value(forKey: allCurrentUsersStorageKey)
+
+        guard let authData = authInfoAny as? Data else {
+            return authInfos
+        }
+
+        let storeAuthInfos = try JSONDecoder().decode([StoreAuthInfo].self, from: authData)
+        for storeAuthInfo in storeAuthInfos {
+            authInfos.append(storeAuthInfo.toAuthInfo)
+        }
+        return authInfos
+    }
+
+    /**
+     * Writes a list of `AuthInfo` structs into some underlying storage.
+     *
+     * - parameters:
+     *     - currentUsersAuthInfo: An array of `AuthInfo` structs to write to `Storage`
+     *     - toStorage: The `Storage` to which to write the `AuthInfo`.
+     * - throws: if the `AuthInfo` could not be encoded into JSON.
+     */
+    internal func writeCurrentUsersAuthInfoToStorage(currentUsersAuthInfo currentUsers: [AuthInfo],
+                                                     toStorage storage: Storage) throws {
+        var currentUsersStore: [StoreAuthInfo] = []
+        for user in currentUsers {
+            currentUsersStore.append(StoreAuthInfo.init(withAuthInfo: user))
+        }
+
+        storage.set(try JSONEncoder().encode(currentUsersStore), forKey: allCurrentUsersStorageKey)
+    }
+
+    /**
+     * Clears the authentication information for the user with the given id.
+     *
+     * - parameters:
+     *     - storage: The `Storage` which should be cleared of any authentication information.
+     */
+    internal func clearUser(storage: Storage, withUserId userId: String?) throws {
```

Comment:
[nit][design] `userId` Should not be a nullable value here. Defaulting to the `activeUserAuthInfo` is an unclear side effect and should be done by the caller, not within this method.